tests/Common/BranchWorkspacesTest.php

```diff
@@ -30,4 +39,124 @@ public function setUp()
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+        $backend->createTable("mytable", ["amount" => ($connection['backend'] === self::BACKEND_SNOWFLAKE) ? "NUMBER" : "VARCHAR"]);
+
+        $tableNames = $backend->getTables();
+        $backend = null; // force odbc disconnect
+
+        $this->assertArrayHasKey("mytable", array_flip($tableNames));
+
+        // get workspace
+        $workspace = $workspaces->getWorkspace($workspace['id']);
+        $this->assertArrayNotHasKey('password', $workspace['connection']);
+
+        // list workspaces
+        $workspacesIds = array_map(function ($workspace) {
+            return $workspace['id'];
+        }, $workspaces->listWorkspaces());
+
+        $this->assertArrayHasKey($workspace['id'], array_flip($workspacesIds));
+
+        $workspaces->deleteWorkspace($workspace['id']);
+
+        // block until async events are processed, processing in order is not guaranteed but it should work most of time
+        $this->createAndWaitForEvent((new \Keboola\StorageApi\Event())->setComponent('dummy')->setMessage('dummy'));
+
+        $events = $this->workspaceSapiClient->listEvents([
+            'runId' => $runId,
+        ]);
+
+        $workspaceCreatedEvent = array_pop($events);
+        $this->assertSame($runId, $workspaceCreatedEvent['runId']);
+        $this->assertSame('storage.workspaceCreated', $workspaceCreatedEvent['event']);
+        $this->assertSame('storage', $workspaceCreatedEvent['component']);
+        $this->assertEquals($this->branchId, $workspaceCreatedEvent['idBranch']);
+
+        $workspaceDeletedEvent = array_pop($events);
+        $this->assertSame($runId, $workspaceDeletedEvent['runId']);
+        $this->assertSame('storage.workspaceDeleted', $workspaceDeletedEvent['event']);
+        $this->assertSame('storage', $workspaceDeletedEvent['component']);
+        $this->assertCredentialsShouldNotWork($connection);
+        $this->assertEquals($this->branchId, $workspaceCreatedEvent['idBranch']);
+    }
+
+    public function testWorkspacePasswordReset()
+    {
+        $workspaces = new Workspaces($this->workspaceSapiClient);
+
+        $workspace = $this->initTestWorkspace();
+
+        $connection = $workspace['connection'];
+
+        $tokenInfo = $this->_client->verifyToken();
+        $this->assertEquals($tokenInfo['owner']['defaultBackend'], $connection['backend']);
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+        $backend->dropTableIfExists('mytable');
+        $backend->createTable("mytable", ["amount" => ($connection['backend'] === self::BACKEND_SNOWFLAKE) ? "NUMBER" : "VARCHAR"]);
+
+        $tableNames = $backend->getTables();
+
+        $this->assertArrayHasKey("mytable", array_flip($tableNames));
+
+        $runId = $this->_client->generateRunId();
+        $this->workspaceSapiClient->setRunId($runId);
```

Comment:
tu je to rovnakej neviem preco tam je v povodnom teste `$this->_client->setRunId($runId);` naviac